packages/gdl-frontend/components/GdlI18nProvider.js

```diff
@@ -6,30 +6,76 @@
  * See LICENSE
  */
 
-import * as React from 'react';
-import { I18nProvider } from '@lingui/react';
+import React, { Component } from 'react';
+import { IntlProvider } from 'react-intl';
+import type { Locale } from '../types';
+import { fetchSiteTranslation } from '../fetch';
+import { setSiteLanguage } from '../lib/storage';
 
-// Currently we only support English
-import catalog from '../locale/en/messages';
+// Currently we initially load English as default and fallback language
+import enTranslations from '../locale/en/en.json';
 
-// For now we have to add each translation here
-const translations = {
-  en: catalog
+const defaultCatalog = {
+  en: enTranslations
 };
 
-// Currently we only support English
-const language = 'en';
+const GdlI18nContext = React.createContext<any>();
+const GdlI18nConsumer = GdlI18nContext.Consumer;
+
+// https://github.com/yahoo/react-intl/wiki/API
+class GdlI18nProvider extends Component<
+  { initialCatalog: *, initialSiteLanguage: string },
+  Locale
+> {
+  state = {
+    language: this.props.initialSiteLanguage,
+    catalog: this.props.initialCatalog
+  };
+
+  componentDidMount() {
+    const { language } = this.state;
+    setSiteLanguage(language);
+  }
+
+  changeSiteLanguage = async (e: { code: string, name: string }) => {
+    const translation = await fetchSiteTranslation(e.code);
+    if (translation.isOk) {
+      // We saves the choosen site language in a cookie
+      setSiteLanguage(e.code);
+      this.setState({
+        catalog: translation.data[e.code],
+        language: e.code
+      });
+    } else {
```

Comment:
Hvis du fetcher et språk som ikke finnes så får du `404` og det er bra! Hva tenker du å sjekke med en `else if (!translation.isOk && translation.statusCode === 404)`? For du kan potensielt får no annet en `404` og da er det kanskje greit å feillogge det med Sentry i en `else {}` setning?

Reply:
Vi kan vel bare sjekke inne i `else` i stedet på `statusCode` for å logge ting som ikke er `404`, tror det blir bedre, tror du ikke?

Reply:
Jo, det høres smartere :D